### PR TITLE
Fix: strict rule supports classes (fixes #2977)

### LINKED ISSUE #2977: Strict mode paradox in ES6 constructor
Here's some ES6 code (I'm using Node 0.10 with ESLint 0.24.0 via gulp-eslint):
```javascript
class Bub extends EventEmitter{
  constructor(config) {
    'use strict';
    this.BASE_URL = 'https://api.telegram.org/bot' + config.token;
    this.TIMEOUT = config.timeout || 864000;
    this.offset = null;
  }
}
```

And here's my `.eslintrc`:
```json
{
  "env": {
    "node": true,
    "es6": true
  },
  "rules": {
    "strict": [2, "function"]
  }
}
```

At the third line, ESLint throws a `no-extra-strict` error. If I remove the statement, it asks me to use the function form of 'use strict'. Is this a bug? Or am I doing something wrong?

### PR DESCRIPTION
**Pull request by mysticatea**
- I followed [this comment](https://github.com/eslint/eslint/issues/2977#issuecomment-121651016), I set `fixes` label.
- I found lacking a check of `ArrowFunctionExpression` on `never` mode and `global` mode, so added those.
- I separated a logic for inside of modules. In this case, all `use strict` directive are unnecessary.
- I added a class support.
- I improved messages of unnecessary.

### CODE REVIEW

**Comment by mysticatea** [comment]
I thought the diff is more understandable when there is `-w` option: https://github.com/eslint/eslint/pull/3028/files?w=1

File: lib/rules/strict.js [hunk 1]
```diff
@@ -17,7 +17,9 @@ var messages = {
     global: "Use the global form of \"use strict\".",
     multiple: "Multiple \"use strict\" directives.",
     never: "Strict mode is not permitted.",
-    unnecessary: "Unnecessary \"use strict\" directive."
+    unnecessary: "Unnecessary \"use strict\" directive.",
+    unnecessaryInModules: "Unnecessary \"use strict\" directive, because inside of modules.",
```
Comment (michaelficarra) [inline lib/rules/strict.js]: Why do these messages need to be different from the regular "unnecessary" message? In a related note, the "global form" vs. the "function form" of a directive don't really mean anything to most users (including me).

Comment (IanVS) [inline lib/rules/strict.js]: > In a related note, the "global form" vs. the "function form" of a directive don't really mean anything to most users (including me).

Are the [docs](http://eslint.org/docs/rules/strict#global-mode) for the rule unclear or confusing?

Comment (michaelficarra) [inline lib/rules/strict.js]: After seeing the documentation, yes, it is more clear what is meant. But I would still consider those the same "form", just in different position. Something like "this directive should be moved to function position" and "this directive should be move to the top level" would be far better. Additionally, the option name "global" doesn't really make sense. The only way "global" is related to "top level" is that variables declared there when the program is parsed as a Script are added to the global object. Anyway, I'd rather not derail this PR, so feel free to open an issue if you agree.

Comment (nzakas) [inline lib/rules/strict.js]: I think the more info we can give people, the better a tool we are.

@mysticatea I would reword these slightly:

1. "use strict" is unnecessary inside of modules
1. "use strict" is unnecessary inside of classes

Comment (mysticatea) [inline lib/rules/strict.js]: @michaelficarra 

> Why do these messages need to be different from the regular "unnecessary" message? 

Because feedbacks of #2785, #2802, and #2957. Developers are not always specialist of the language spec, so those might be confused by just `unnecessary`.

@nzakas Thank you for correction!

**Comment by ilyavolodin** [comment]
LGTM
